### ISSUE TITLE
Event listener

### ISSUE THREAD

**Issue by sepik** [perlundq/yajsync#50]
It would be great to add event listener to client API in order to track rsync progress.
-v console output may be done from listener implementation.

**Comment by perlundq** [comment]
Yes something like that would definitely be useful (for file related events). Will have to think about the design of it though.

**Comment by whybeyoung** [comment]
any progress here?  it's really need a rsync progress when uploading/downloading.....
